lightning/src/ln/channelmanager.rs

```diff
@@ -3044,6 +3044,56 @@ impl<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelManager<M, T, K, F
 		Ok(())
 	}
 
+	/// Attempts to forward an intercepted HTLC over the provided channel id and with the provided
+	/// amount to forward. Should only be called in response to an [`HTLCIntercepted`] event.
+	///
+	/// Intercepted HTLCs can be useful for Lightning Service Providers (LSPs) to open a just-in-time
+	/// channel to a receiving node if the node lacks sufficient inbound liquidity.
+	///
+	/// To make use of intercepted HTLCs, use [`ChannelManager::get_intercept_scid`] to generate short
+	/// channel id(s) to put in the receiver's invoice route hints. These route hints will signal to
+	/// LDK to generate an [`HTLCIntercepted`] event when it receives the forwarded HTLC.
+	///
+	/// Note that LDK does not enforce fee requirements in `amt_to_forward_msat`, and will not stop
+	/// you from forwarding more than you received.
+	///
+	/// [`HTLCIntercepted`]: events::Event::HTLCIntercepted
+	// TODO: when we move to deciding the best outbound channel at forward time, only take
+	// `next_node_id` and not `next_hop_channel_id`
+	pub fn forward_intercepted_htlc(&self, intercept_id: InterceptId, next_hop_channel_id: &[u8; 32], _next_node_id: PublicKey, amt_to_forward_msat: u64) -> Result<(), APIError> {
```

Comment:
just a question to check my understanding:

Is this basically the procedure for an LDK user?

1. Calls `ChannelManager::get_intercept_scid`
2. Generates invoice with intercept scid in route hints for end-user
3. Receives an `HTLCIntercepted` event when the HTLC is intercepted
4. At this point the LDK user can create a channel
5. Then call this method to forward over that JIT channel

Reply:
Yes that's my understanding, where the LDK user is the LSP in your example (`nodes[1]` in the test), and the end-user is `nodes[2]` in the test. I guess step (2) in your example doesn't necessarily need to be on the LDK users side, as long as the Intercept scid is passed to the end-user.

Reply:
Ah, I should have read the test. When you say.

Yeah I think I phrased that weirdly for (2). End user generates invoice but just shoves the scid in the route hints.